Dshop/apps/products_catalogue/serializers.py

```diff
@@ -1,10 +1,67 @@
+from django.db import transaction
+from django.shortcuts import get_object_or_404
+from dj_shop_cart.cart import get_cart_class
 from rest_framework import serializers
 from .models import Product
 
-
 class ProductSerializer(serializers.ModelSerializer):
-
     class Meta:
         model = Product
         fields = ['id', 'category', 'name', 'price', 'short_description', 'full_description', 'parent_product']
         read_only_fields = ['id']
+
+
+class CartItemSerializer(serializers.Serializer):
+    product_pk = serializers.IntegerField(min_value=1, required=True)
+    quantity = serializers.IntegerField(min_value=1, required=True)
+    product_name = serializers.CharField(read_only=True, max_length=200)
+    price = serializers.DecimalField(
+        read_only=True, min_value=0, max_digits=10, decimal_places=2
+    )
+    subtotal = serializers.DecimalField(
+        read_only=True, min_value=0, max_digits=10, decimal_places=2
+    )
+    item_id = serializers.IntegerField(read_only=True, min_value=1)
+    
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        product_name = Product.objects.get(pk=instance.product_pk).name
+        representation['product_name'] = product_name
+        return representation
+
+
+class CartReadSerializer(serializers.Serializer):
+    total = serializers.DecimalField(
+        read_only=True, min_value=0, max_digits=10, decimal_places=2
+    )
+    items = serializers.SerializerMethodField()
+    count = serializers.IntegerField(read_only=True, min_value=1)
+
+    def get_items(self, obj):
+        return CartItemSerializer(list(obj), many=True).data
+
+
+class CartWriteSerializer(serializers.Serializer):
+    items = CartItemSerializer(many=True, required=False)
+
+    def create(self, validated_data):
+        request = self.context['request']
+        cart = get_cart_class().new(request)
+        cart.empty()
+        with transaction.atomic():
```

Comment:
Actually, can you simply turn on atomic_requests in settings.py? Then this is not necessary.
https://docs.djangoproject.com/en/5.0/topics/db/transactions/#tying-transactions-to-http-requests

Reply:
imho global only atomic requests would be an overkill. It's probably the only place with `transaction.atomic` in Dshop

Reply:
Your call. But, atomic requests are great thing to tie HTTP request to transaction which makes system much more resilient. Slower, with bigger traffic, but we don't have traffic :P